lazy_dataset/core.py

```diff
@@ -1640,26 +1757,69 @@ def indexable(self):
     def __iter__(self):
         buckets = list()
         for i, sample in enumerate(self.input_dataset):
-            value = min(self.key(sample), self.max_value)
+            value = self.key(sample)
             found_bucket = False
-            for i, (bucket, min_value, max_value) in enumerate(buckets):
-                if min_value <= value <= max_value:
+            expired = set()
+            for j, (
+                bucket, creation_idx, lower_bound, upper_bound, max_value
+            ) in enumerate(buckets):
+                if (
+                    self.expiration is not None
+                    and (i - creation_idx) / self.batch_size > self.expiration
+                ):
+                    expired.add(j)
+                    if not self.drop_incomplete:
+                        if self.sort_by_key:
+                            bucket = sorted(bucket, key=self.key, reverse=True)
+                        yield bucket
```

Comment:
Is it intended, that the "sample" cannot be appended to expired buckets? (`lower_bound <= value <= upper_bound` can be true for an `expired` bucket)